generator/src/lib.rs

```diff
@@ -1822,25 +1822,16 @@ fn derive_debug(
         let param_ident = field.param_ident();
         let param_str = param_ident.to_string();
         let debug_value = if is_static_array(field) && field.basetype == "char" {
-            quote! {
-                &unsafe {
-                    ::std::ffi::CStr::from_ptr(self.#param_ident.as_ptr())
-                }
-            }
+            let param_ident = format_ident!("{}_as_c_str", param_ident);
+            quote!(&unsafe { self.#param_ident() })
```

Comment:
I think this is actually unsound: the user can safely construct a structure with non-null-terminated data in a string field. We need a checked version for `Debug`. Not sure it's worth the trouble to preserve an unchecked version along side.

Reply:
It has always been unsound as written. This PR is not changing those semantics, #746 is taking care of making that safe.

We can also make a separate contribution to address those soundness issues by means of cloning the implementation of `CStr::from_bytes_until_nul()` for a slice, but (as written in https://github.com/ash-rs/ash/pull/746#issuecomment-1826270501) I'd rather bump the MSRV instead.